pkg/kvstore/consul_test.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2016-2017 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kvstore
+
+import (
+	"io"
+	"net/http"
+	"os"
+	"testing"
+
+	consulAPI "github.com/hashicorp/consul/api"
+)
+
+var handler http.HandlerFunc
+
+func TestMain(m *testing.M) {
+	mux := http.NewServeMux()
+	// path is hardcoded in consul
+	mux.HandleFunc("/v1/status/leader", func(w http.ResponseWriter, r *http.Request) {
+		handler(w, r)
+	})
+
+	srv := &http.Server{
+		Addr:    ":8000",
+		Handler: mux,
+	}
+
+	go srv.ListenAndServe()
+	defer srv.Close()
+
+	os.Exit(m.Run())
+}
+
+func TestConsulClientRetry(t *testing.T) {
+	triesAttempted := 0
+	maxRetries = 3
+
+	handler = func(w http.ResponseWriter, r *http.Request) {
+		triesAttempted++
+		http.Error(w, "retry test error", http.StatusInternalServerError)
+	}
+
+	_, err := NewConsulClient(&consulAPI.Config{
+		Address: ":8000",
+	})
+
+	// we should get a failure
+	if err == nil {
+		t.Log("no error")
+		t.FailNow()
+	}
+
+	if triesAttempted == 0 || maxRetries != triesAttempted {
+		t.Logf("expected %d retries, got %d", maxRetries, triesAttempted)
+		t.FailNow()
+	}
+}
+
+func TestConsulClientOk(t *testing.T) {
+	triesAttempted := 0
+	maxRetries = 3
+
+	handler = func(w http.ResponseWriter, r *http.Request) {
+		triesAttempted++
+		io.WriteString(w, "\"nanananananananananleaaderrrr\"")
+	}
```

Comment:
What does `nanananananananananleaaderrrr` mean? Please change it to something more meaningful.